### PR TITLE
Fee and Trades Improvements

### PR DESCRIPTION
**Pull request by sambarboza**

### CODE REVIEW

File: contracts/trade/src/contract.rs [hunk 1]
```diff
@@ -202,59 +253,53 @@ fn try_release(
...
+    let mut send_msgs: Vec<SubMsg> = Vec::new();
+    let balance = balance_result.unwrap();
 
-    let final_balance = balance.clone();
+    let mut final_balance: Vec<Coin> = Vec::new();
+    let offer = get_offer(&deps.as_ref(), &state);
 
-    //Query maker to send to the incentives contract
-    let offer: Offer = deps
+    let factory_cfg: FactoryConfig = deps
```
[the remark this answers is not in the document]
Comment (sambarboza) [reply contracts/trade/src/contract.rs]: Good catch! I forgot that we had this method.

File: contracts/trade/src/contract.rs [hunk 2]
```diff
@@ -283,25 +328,22 @@ fn get_ust_amount(info: MessageInfo) -> Uint128 {
...
-        .amount
-        .checked_div(Uint128::new(1000))
-        .unwrap();
-    Ok([Coin {
-        amount: fee_amount,
-        denom: balance[0].clone().denom,
-    }]
-    .to_vec())
+pub fn subtract_localterra_fee(amount: Uint128) -> Uint128 {
+    amount.clone().checked_div(Uint128::new(1000u128)).unwrap()
```
[the remark this answers is not in the document]
Comment (sambarboza) [reply contracts/trade/src/contract.rs:326]: You can initialize an Uint128 with a string or a native integer, the native integer is the u128 type.

File: packages/protocol/src/trade.rs [hunk 3]
```diff
@@ -26,6 +27,7 @@ pub enum QueryMsg {
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct State {
+    pub addr: Addr,
```
[the remark this answers is not in the document]
Comment (sambarboza) [reply packages/protocol/src/trade.rs:30]: Thank you. Done @ 0fea23daf1cf40745490fe894105fda7749c987b

**Review by davirds — APPROVED** [review]
LGTM!

**Pull request merged**